saleor/plugins/avatax/plugin.py

```diff
@@ -305,19 +305,50 @@ def calculate_checkout_line_total(
             return base_total
 
         taxes_data = get_checkout_tax_data(checkout_info, lines, discounts, self.config)
+        return self._calculate_total_price(
+            taxes_data, checkout_line_info.variant.sku, previous_value
+        )
+
+    def calculate_order_line_total(
+        self,
+        order: "Order",
+        order_line: "OrderLine",
+        variant: "ProductVariant",
+        product: "Product",
+        previous_value: TaxedMoney,
+    ) -> TaxedMoney:
+        if self._skip_plugin(previous_value):
+            return previous_value
+
+        if not variant or (variant and not product.charge_taxes):
```

Comment:
The typing for calculate_order_line_total doesn't assume that you will receive Optional variant here.

Reply:
Also, I would say that `if not variant or not product.charge_taxes:` should be enough here 🤔